common/lib/xmodule/xmodule/js/spec/video/sjson_spec.js

```diff
@@ -23,12 +25,38 @@ function (Sjson) {
         });
 
         it('search returns a correct caption index', function () {
-            expect(sjson.search(0)).toEqual(-1);
-            expect(sjson.search(3120)).toEqual(1);
-            expect(sjson.search(6270)).toEqual(2);
-            expect(sjson.search(8490)).toEqual(2);
-            expect(sjson.search(21620)).toEqual(4);
-            expect(sjson.search(24920)).toEqual(5);
+            expect(sjson.search(video_stops[0])).toEqual(0);
+            expect(sjson.search(video_stops[1])).toEqual(1);
+            expect(sjson.search(video_stops[2])).toEqual(2);
+            expect(sjson.search(video_stops[3])).toEqual(2);
+            expect(sjson.search(video_stops[4])).toEqual(4);
+            expect(sjson.search(video_stops[5])).toEqual(5);
+        });
+
+        it('search returns the last entry for a value outside the bounds of the array', function() {
+            expect(sjson.search(OUT_OF_BOUNDS_STOP)).toEqual(sjson.getCaptions().length - 1);
+        });
+
+        it('search only searches through a subrange of times if start / end times are specified', function () {
+            var start = video_stops[2] - 100;
```

Comment:
What do you think of changing `video_stops` to a dict instead of an array?

Being able to index on keys will make a lot of this code more transparent.

It's difficult and requires rescanning the document to know what `video_stops[2]` means.

It's much easier to decipher entries like:
- `video_stops['HALFWAY_THROUGH']`
- or even
- `video_stops['NEAR_THE_BEGINNING']`